posts/2022-07-12-components-with-dynamic-attributes.md

```diff
@@ -0,0 +1,272 @@
+%{
+  author: "Tracey Onim",
+  author_link: "https://github.com/TraceyOnim",
+  tags: ["LiveView", "Components", "LiveView Helpers"],
+  date: ~D[2022-07-12],
+  title: "Components with dynamic attributes",
+  excerpt: """
+  Learn how you can support dynamic attributes when using reusable components with assigns_to_attribute/2 function.
+  """
+}
+---
+
+
+I’m quite sure most of you, have already interacted with components you have created more components to reuse in your application.However, sometimes you may feel stuck when creating reusable components more so when you want components to meet certain criteria. For example, you may want the feature in component A to be different from that of component B.
+
+What exactly I’m I trying to say? Lets say we are creating an application akin to kanban. I want us to create an application that can help users plan their weekly task.
+
+
+![board plan](/images/board-plan.png)
+
+
+From the above image , I have split the board into 3 columns. That is the house, work and school columns. The user interacting with this board is supposed to add weekly task they are supposed to do in each sector.
+
+You will also notice that each sector has a card of different color. I want us to use reusable component to implement in our application. The board will act as the parent LiveView while the card will be the component.
+
+Now that we want the cards in each section to have different colors, how can we make it possible keeping in mind that we are using reusable components.
+
+
+Let’s begin:
+
+## 1.Create Board LiveView
```

Comment:
It should be "live view" when referring to a single live view page/process. Use "LiveView" when referring to the framework.